fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/ingame/TransferTestInitializer.java

```diff
@@ -63,6 +74,74 @@ public void onInitialize() {
 		ItemStorage.SIDED.registerForBlocks((world, pos, state, be, direction) -> TrashingStorage.ITEM, Blocks.OBSIDIAN);
 		// And diamond ore blocks are an infinite source of diamonds! Yay!
 		ItemStorage.SIDED.registerForBlocks((world, pos, state, be, direction) -> CreativeStorage.DIAMONDS, Blocks.DIAMOND_ORE);
+
+		CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> {
+			dispatcher.register(
+					CommandManager.literal("fabric_insertintoheldstack")
+							.then(CommandManager.argument("stack", ItemStackArgumentType.itemStack(registryAccess))
+									.then(CommandManager.argument("count", LongArgumentType.longArg(1))
+											.executes(context -> {
+												ItemVariant variant = ItemVariant.of(ItemStackArgumentType.getItemStackArgument(context, "stack")
+														.createStack(1, false));
+
+												ContainerItemContext containerCtx = ContainerItemContext.ofPlayerHand(context.getSource().getPlayerOrThrow(), Hand.MAIN_HAND);
+												Storage<ItemVariant> storage = ItemStorage.ITEM.find(containerCtx.getItemVariant().toStack(), containerCtx);
+
+												if (storage == null) {
+													context.getSource().sendMessage(Text.literal("no storage found"));
+													return 0;
+												}
+
+												long inserted;
+
+												try (Transaction tx = Transaction.openOuter()) {
+													inserted = storage.insert(
+															variant,
+															LongArgumentType.getLong(context, "count"),
+															tx
+													);
+													tx.commit();
+												}
+
+												context.getSource().sendMessage(Text.literal("inserted " + inserted + " items"));
+
+												return (int) inserted;
+											})))
+			);
+
+			dispatcher.register(
+					CommandManager.literal("fabric_extractfromheldstack")
+							.then(CommandManager.argument("stack", ItemStackArgumentType.itemStack(registryAccess))
+									.then(CommandManager.argument("count", LongArgumentType.longArg(1))
+											.executes(context -> {
+												ItemVariant variant = ItemVariant.of(ItemStackArgumentType.getItemStackArgument(context, "stack")
```

Comment:
Is there any check against trying to extract blank? Or maybe allowing that is a feature?